### PR TITLE
Add pathlib.Path to string conversion

### PR DESCRIPTION
**Pull request by bakyazi**
#2642 
Check path parameters that is pathlib.Path instance. If it is, convert it to string.

### CODE REVIEW

File: nilearn/_utils/numpy_conversions.py [hunk 1]
```diff
@@ -149,7 +150,7 @@ def csv_to_array(csv_path, delimiters=' \t,;', **kwargs):
     array: numpy.ndarray
         An array containing the data loaded from the CSV file.
     """
-    if not isinstance(csv_path, str):
+    if not isinstance(csv_path, str) or not isinstance(csv_path, pathlib.Path):
```
Comment (jeromedockes) [inline nilearn/_utils/numpy_conversions.py:153]: the `or` should maybe be `and` ?

Comment (bakyazi) [reply nilearn/_utils/numpy_conversions.py]: yes you are right. i am fixing

**Comment by jeromedockes** [comment]
thanks for working on this @bakyazi !

**Review by bthirion — COMMENTED** [review]
Thx for tackling this !
Please add a whatsnew entry... and tests for all these additions...

**Comment by GaelVaroquaux** [comment]
Did you close this by mistake?

My gut feeling is to reopen it, so that it doesn't get lost.

Thanks!!

**Comment by bakyazi** [comment]
hello actually because i couldn't handle unittests, i closed pr. i could not figure out existing unittest to add mine. But if you want, you can take up it.

**Comment by NicolasGensollen** [comment]
@bakyazi I can help with the tests if you want to reopen...

**Pull request closed without merging**